R/brreg-utils.R

```diff
@@ -0,0 +1,87 @@
+#' null2na
+#' @noRd
+null2na <- function(x) if (is.null(x) || length(x) == 0) NA else x
+
+#' request_brreg
+#' @noRd
+request_brreg <- function(type = c('enheter', 'kommuner'), entity = NULL) {
+  type <- match.arg(type)
+  req <- request(brreg_url)
+  if (type == 'enheter') {
+    is_valid_number <- grepl('\\d{9}', entity)
+    #
+    if (nchar(entity) != 9 && !grepl('\\D', entity) ) { #grepl('\\d', entity) && grepl('\\D', entity)
+      msg = '{entity} is not a 9-digit organization number'
+      cli::cli_abort(msg)
+    }
+    if (is_valid_number) {
+      req <- req |>
+        req_url_path_append('enheter') |>
+        req_url_path_append(entity)
+    } else {
+      req <- req |>
+        req_url_path_append('enheter') |>
+        req_url_query('navn' = entity)
+    }
+  } else if (type == 'kommuner') {
+    req <- req |>
+      req_url_path_append('kommuner') |>
+      req_url_query(size = 450)
+  }
+  send_query(req)
+}
+
+#' parse_brreg_entity
+#' @noRd
+parse_brreg_entity <- function(parsed) {
+  if ('organisasjonsnummer' %in% names(parsed)) {
+    df <- parse_brreg_entity_(parsed)
+  } else if ('_embedded' %in% names(parsed)) {
+    # NOTE: Old code from riksrevR. Not sure this is needed.
+    # if (length(parsed$`_embedded`$enheter) == 0) {
+    #   cli::cli_warn('No entities found.')
+    #   return(invisible(NULL))
+    # }
+    dl <- lapply(parsed$`_embedded`$enheter, parse_brreg_entity_)
+    df <- do.call('rbind', dl)
+  } else if (length(names(parsed) == 2)) { # "_links" "page"
+    cli::cli_warn('No entities found.')
+    return(invisible(NULL))
+  }
+  df
+}
+
+#' parse_brreg_entity_
+#' @noRd
+parse_brreg_entity_ <- function(p) {
+
+  # Create data.frame
+  data.frame(
```

Comment:
Consider consistent use of `_` or `.` as separator

Reply:
Yep. Not sure about the best convention here. I sometimes use `*_single` (e.g.  `parse_brreg_entity_single`) for these kind of functions, or we could use `.parse_brreg_entity()`, but ending with just an extra `_` (current) is also not that uncommon.

Reply:
The original comment pertained to the column names in the data.frame.

Reply:
Ah. I see. Will definitely look at that. This is just a copy-paste from the old `{riksrevr}` code.

Reply:
This has been fixed